### ISSUE TITLE
Alfred Spotify Mini Player randomly forgets Client ID credentials

### ISSUE THREAD

**Issue by chrismessina** [vdesabou/alfred-spotify-mini-player#439]
![CleanShot 2020-12-29 at 20 02 18@2x](https://user-images.githubusercontent.com/22610/103329275-ed885880-4a10-11eb-8514-859eba6fd932.png)

Alfred just popped up in the middle of me doing something else with this screen — complaining that "Your application Client ID is missing", even though I previously successfully setup the Client ID and secret, and authorized the app against Spotify.

Why did it do this? How can I debug this?

**Comment by vdesabou** [comment]
The only time I'm doing it in the code is [here](https://github.com/vdesabou/alfred-spotify-mini-player/blob/16a6887dc37b8a91008a1876b5f57e6eac6e49f6/spotify-mini-player/src/refreshLibrary.php#L231-L232), it's during a refresh of library when some permission scope is missing
To investigate further, can you send me a "debug email" by calling "spot_mini_debug" and choosing first option

**Comment by chrismessina** [comment]
I ran the command, but an email was not generated. In the Alfred console, I saw this error:

`[15:30:34.348] STDERR: Spotify Mini Player[Run Script] The file /Users/messina/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.EBFE4415-0621-45C7-A1D7-F3CD4F75FDFE/mailto:alfred.spotify.mini.player@gmail.com?subject=Alfred Spotify Mini Player debug file (version 10.4)&body=ℹ️ Please take time to describe your problem in a few lines 👇`

It appears that an email can not be generated this way. It might be better to pass the output of this to a Workflow Object (i.e. Open URL)?

I'll gather the debug files generated and send them manually via email.

**Comment by vdesabou** [comment]
@chrismessina It is probably because you don't have default email app defined at os level. The information is also available in your clipboard, so even if opening the email app is not working, you can still share manually by email. I saw your email with logs, but I would be interested to see the complete debug information, can you send another email with content of clipboard ?

**Comment by chrismessina** [comment]
Sent!

**Comment by vdesabou** [comment]
@chrismessina It's still unclear from the logs what happened exactly, and I was wrong earlier, there is nothing in the code that delete `oauth_client_id` setting which corresponds to client id credential.

The only thing that could have happened is that all your settings were reset. But again, that can only happen when executing `spot_mini_reset_settings` command.

This is probably not what happened in your case.

Did it happen only once ?

**Comment by chrismessina** [comment]
> Did it happen only once ?

It happened 2-3 times, which is why I reported it. It hasn't happened since, but I think I also disabled Spotificious once I started using your Workflow. I have no idea if they're related, but that's the only other thing I remember changing around that time. 

Also — the album artwork download process did take many days to complete, so I wonder if Spotify somehow throttled my requests? Really, I have no idea — but it seems like the issue hasn't repeated since the album artwork download process completed.
